src/contrib/cluster/Akka.Cluster.Tools.Tests/Singleton/OldChangedBufferSpecs.cs

```diff
@@ -0,0 +1,151 @@
+//      Copyright (C) 2009-2024 Lightbend Inc. <http://www.lightbend.com>
+//      Copyright (C) 2013-2024 .NET Foundation <https://github.com/akkadotnet/akka.net>
+//  </copyright>
+// -----------------------------------------------------------------------
+
+#nullable enable
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading.Tasks;
+using Akka.Actor;
+using Akka.Cluster.Tools.Singleton;
+using Akka.Configuration;
+using Akka.TestKit;
+using FluentAssertions;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Akka.Cluster.Tools.Tests.Singleton;
+
+/// <summary>
+/// Reproduction for https://github.com/akkadotnet/akka.net/issues/7196 - clearly, what we did
+/// </summary>
+public class OldChangedBufferSpecs : AkkaSpec
+{
+    private readonly ActorSystem _hostNodeV1;
+    private readonly ActorSystem _otherNodeV2;
+
+    private static Config OriginalNodeConfig() => """
+                                                  
+                                                                akka.loglevel = INFO
+                                                                akka.actor.provider = "cluster"
+                                                                akka.cluster.roles = [non-singleton]
+                                                                akka.cluster.auto-down-unreachable-after = 2s
+                                                                akka.cluster.singleton.min-number-of-hand-over-retries = 5
+                                                                akka.cluster.app-version = "1.0.0"
+                                                                akka.remote {
+                                                                  dot-netty.tcp {
+                                                                    hostname = "127.0.0.1"
+                                                                    port = 0
+                                                                  }
+                                                                }
+                                                  """;
+
+    private static Config V2NodeConfig(ActorSystem originalSys) => ConfigurationFactory.ParseString(
+        "akka.cluster.app-version = \"1.0.2\"").WithFallback(originalSys.Settings.Config);
+
+    public OldChangedBufferSpecs(ITestOutputHelper output) : base(OriginalNodeConfig(), output)
+    {
+        _hostNodeV1 = ActorSystem.Create(Sys.Name,
+            ConfigurationFactory.ParseString("akka.cluster.roles = [singleton]").WithFallback(Sys.Settings.Config));
+        _otherNodeV2 = ActorSystem.Create(Sys.Name,
+            ConfigurationFactory.ParseString("akka.cluster.roles = [singleton]").WithFallback(V2NodeConfig(Sys)));
+    }
+
+    [Fact(DisplayName =
+        "Singletons should not move to higher AppVersion nodes until after older incarnation is downed")]
+    public async Task Bugfix7196Spec()
```

Comment:
End to End reproduction spec:

1. Forms 2 node cluster: 1 with the role enabled for singleton (` _hostNodeV1`), the other without (`Sys`). Both nodes are running with `AppVersion` == 1.0.0.
1. `Sys` will create the `ClusterSingletonProxy` - which will point to the `SingletonManger` and the resulting `Singleton` actor hosted on `_hostNodeV1`.
1. `_otherNodeV2` joins the cluster with `AppVersion` == 1.0.2. The bug from #7196 meant that this node would IMMEDIATELY become the `Oldest` and the singleton would move (and the old one would not die, creating a split brain.)
1. We validate that the singleton is still on `_hostNodeV1` - which means we've fixed the bug, but we still need to make sure the oldest gets replaced if the current one leaves.
1. We down `_hostNodeV1` and observe that the singleton migrates over to `_otherNodeV2` afterwards.

This validates, end to end, that the issue is resolved. The only way we could make this more robust is adding a second `_hostNodeV1` instance and observe that `_otherNodeV2` gets elected in favor over it, but we have other tests below that ensure this.